src/pages/library/common.js

```diff
@@ -86,8 +174,74 @@ const FilterSection = ({ onTagFilter, labels, selectedTags, labelRenderer, listD
     }, labelsToDisplay),
     _.size(labels) > numLabelsToRender && h(Link, {
       style: { display: 'block', textAlign: 'center' },
-      onClick: () => setShowAll(!showAll)
-    }, [`See ${showAll ? 'less' : 'more'}`])
+      onClick: () => {
+        setShowAll(!showAll)
+        setFilterChanges({})
+
+        const _tagFilters = {}
+        _.forEach(label => {
+          const lowerTag = _.toLower(label)
+          _tagFilters[lowerTag] = listDataByTag[lowerTag]
+        }, labels)
+        setFilteredTags(_tagFilters)
+      }
+    }, [`See more`]),
+    showAll && h(Modal, {
+      title: ``,
+      width: '100%',
+      showButtons: true,
+      okButton: 'Apply Filters',
+      styles: {
+        modal: { maxWidth: 900, padding: 30 },
+        buttonRow: { width: '100%', borderTop: `6px solid ${colors.dark(0.1)}`, paddingTop: 20 }
+      },
+      onDismiss: () => {
+        setShowAll(false)
+        _.forEach(onTagFilter, filterChanges)
+        setFilterChanges({})
+        setFilteredLabels(labels)
+      }
+    }, [
+      h2({ style: { marginTop: 0 } }, [`Filter by: "${name}"`]),
+      h(FilterBar, {
+        name,
+        onClear: () => { setFilteredLabels(labels) },
+        onSortBy: sort => {
+          if (sort === 'top') {
+            setFilteredLabels(_.sortBy(label => -1 * _.size(filteredTags[_.toLower(label)]), labels))
+          }
+        },
+        onFilterByLetter: letter => { setFilteredLabels(labelsByFirstChar[letter]) },
+        onFilterBySearchText: searchText => {
+          setFilteredLabels(_.filter(label => _.toLower(label).match(_.escapeRegExp(_.toLower(searchText))), labels))
+        }
+      }),
+      div({ style: { height: 'calc(80vh - 250px)', minHeight: 300, overflowY: 'auto' } }, [
+        div({
+          style: {
+            display: 'flex', flexDirection: 'row', flexWrap: 'wrap', justifyContent: 'space-between',
+            fontSize: '1rem', textTransform: 'capitalize', marginTop: 20
+          }
+        }, _.map(label => {
+          const lowerTag = _.toLower(label)
+          return div({ className: 'label', style: { width: '25%', margin: '0 15px 10px 30px', position: 'relative', minHeight: 30 } }, [
+            h(LabeledCheckbox, {
+              checked: !!filterChanges[lowerTag] ^ _.includes(lowerTag, lowerSelectedTags),
+              onChange: v => {
+                filterChanges[lowerTag] ? delete filterChanges[lowerTag] : filterChanges[lowerTag] = { lowerTag, label }
+                setFilterChanges({ ...filterChanges })
+              },
```

Comment:
We tend to avoid mutation where we can, especially on react state values. Mutating the state outside of a set state could have unintended effects.

The react `useState` hook allows us to set state with a function, where the parameter passed in is the previous state. I think this will do the same thing without the mutation:

```suggestion
onChange: () => {
                setFilterChanges(prevFilter => prevFilter[lowerTag] ? _.omit(lowerTag, prevFilter) : _.set(lowerTag, { lowerTag, label }, prevFilter))
              },
```